inference-engine/tests/functional/inference_engine/transformations/compare_functions_test.cpp

```diff
@@ -208,4 +208,127 @@ TEST(TransformationTests, CompareFunctoinsTINegative) {
     auto res = compare_functions(f, f_ref);
     EXPECT_FALSE(res.first);
     EXPECT_EQ(res.second, "LSTMCell/4 != Relu/0");
-}
\ No newline at end of file
+}
+
+static inline bool contains(const std::string& text, const std::string& search_for) {
+    return text.find(search_for) != std::string::npos;
+}
+
+TEST(TransformationTests, ConstantNegativeDifferentElementType) {
+    const auto& f1 = [] {
+        using namespace ngraph::opset5;
+        auto constant = Constant::create(ngraph::element::f64, ngraph::Shape{1}, {10.1});
+
+        return std::make_shared<ngraph::Function>(
+            ngraph::NodeVector{constant}, ngraph::ParameterVector{});
+    }();
+
+    const auto& f2 = [] {
+        using namespace ngraph::opset5;
+        auto constant = Constant::create(ngraph::element::f32, ngraph::Shape{1}, {10.1});
+
+        return std::make_shared<ngraph::Function>(
+            ngraph::NodeVector{constant}, ngraph::ParameterVector{});
+    }();
+
+    const auto& res = compare_functions(f1, f2, false, false, false, true, true);
+    EXPECT_FALSE(res.first);
+    EXPECT_TRUE(contains(res.second, "mismatch in value: 'element_type'"))
```

Comment:
You can use EXPECT_HAS_SUBSTRING from google tests.
```suggestion
    EXPECT_HAS_SUBSTRING(res.second, "mismatch in value: 'element_type'")
```

Reply:
Will fix.

Reply:
`EXPECT_HAS_SUBSTRING` is part of ngraph test it can't be easy include in `inference-engine` part.
I use gMock instead ngraph extension.  It looks like this ` EXPECT_THAT(res.second, HasSubstr("mismatch in value: 'element_type'"));` 
-> https://chenchang.gitbooks.io/googletest_docs/content/googlemock/CheatSheet.html